usersync/cookie.go

```diff
@@ -162,9 +215,31 @@ func (cookie *PBSCookie) GetId(bidderName openrtb_ext.BidderName) (id string, ex
 // SetCookieOnResponse is a shortcut for "ToHTTPCookie(); cookie.setDomain(domain); setCookie(w, cookie)"
 func (cookie *PBSCookie) SetCookieOnResponse(w http.ResponseWriter, domain string, ttl time.Duration) {
 	httpCookie := cookie.ToHTTPCookie(ttl)
+	var MAX_COOKIE_SIZE int = 1 << 15 // 32 KB
+	var now time.Time = time.Now()
+
 	if domain != "" {
 		httpCookie.Domain = domain
 	}
+
+	var currSize int = len([]byte(httpCookie.String()))
+	for currSize > MAX_COOKIE_SIZE {
+		var oldestElem string = ""
+		var oldestDate int64 = math.MaxInt64
+		for key, value := range cookie.uids {
+			var timeSince time.Duration = now.Sub(value.Expires)
```

Comment:
You shouldn't need to specify `time.Duration` as that is already defined as the return type of `now.Sub()`

Reply:
Also, since you are taking the current time, and subtracting the expirery time, this will result in a negative number. The most negative number will be the expires time that is furthest from now, so in the end we will remove the entry that has the longest time before it expires.

Reply:
Gosh, I got the `Sub()` function backwards, corrected